### PR TITLE
Added n_features_in_ to three of the embedded class estimators

### LINKED ISSUE #352: Add `n_features_in_` for scikit-learn compliance
`FutureWarning: As of scikit-learn 0.23, estimators should expose a n_features_in_ attribute, unless the 'no_validation' tag is True. This attribute should be equal to the number of features passed to the fit method. An error will be raised from version 0.25 when calling check_estimator().`

### PR DESCRIPTION
**Pull request by zeou1**
#### Reference Issues/PRs

Fixes #352 

#### What does this implement/fix? Explain your changes.
Added n_features_in_ to three embedded class estimators, ase.py, lse.py, and mds.py
For ase and lse, the n_features_in_ attribute equals to the number of points in the graph. And for mds, the attribute equals to the number of features of each sample.

### CODE REVIEW

File: graspy/embed/mds.py [hunk 1]
```diff
@@ -203,6 +206,10 @@ def fit(self, X, y=None):
         self.components_ = U
         self.singular_values_ = D ** 0.5
         self.dissimilarity_matrix_ = dissimilarity_matrix
+        if X.ndim == 2:
+            self.n_features_in_ = X.shape[1]
+        else:
+            self.n_features_in_ = X.shape[1] * X.shape[2]
```
[the remark this answers is not in the document]
Comment (bdpedigo) [reply graspy/embed/mds.py:212]: i think so

File: graspy/embed/mds.py [hunk 2]
```diff
@@ -203,6 +206,10 @@ def fit(self, X, y=None):
         self.components_ = U
         self.singular_values_ = D ** 0.5
         self.dissimilarity_matrix_ = dissimilarity_matrix
+        if X.ndim == 2:
```
Comment (asaadeldin11) [inline graspy/embed/mds.py:209]: No need to check `X.ndim`. A single line `self.n_features_in_ == X.shape[1]` should do.

File: graspy/embed/mds.py [hunk 3]
```diff
@@ -71,6 +71,9 @@ class ClassicalMDS(BaseEstimator):
         Equals the parameter n_components. If input n_components was None,
         then equals the optimal embedding dimension.
 
+    n_features_in_: int, or array, shape(1, 2)
```
Comment (asaadeldin11) [inline graspy/embed/mds.py:74]:
```suggestion
    n_features_in_: int
```

File: graspy/embed/mds.py (hunk 1 above)
Comment (asaadeldin11) [inline graspy/embed/mds.py:212]:
```suggestion
```

**Review by asaadeldin11 — CHANGES_REQUESTED** [review]
@zeou1 please take a look at the above comments

**Comment by zeou1** [comment]
Closing the PR due to the merge with Microsoft. I forked a new repo and made a new PR request here: #462 . All future conversation can be continued there.

**Pull request closed without merging**